spec/ddtrace/contrib/rails/rails_auto_instrument_spec.rb

```diff
@@ -0,0 +1,41 @@
+require 'ddtrace/contrib/rails/rails_helper'
+
+RSpec.describe 'Datadog::AutoInstrument' do
+  include_context 'Rails test application'
+
+  around do |example|
+    # Reset before and after each example; don't allow global state to linger.
+    Datadog.configuration.reset!
+    example.run
+    Datadog.configuration.reset!
+  end
+
+  context 'when auto patching is included' do
+    before do
+      skip 'Java not supported' if RUBY_PLATFORM == 'java'
+      allow(ENV).to receive(:[]).and_call_original
+      allow(ENV).to receive(:[]).with('TEST_AUTO_INSTRUMENT').and_return(true)
```

Comment:
`ClimateControl` allows us to mock environment variables, without having to worry if the exercises code is calling `ENV[]`, `ENV.fetch`, or `ENV.each`.
https://github.com/DataDog/dd-trace-rb/blob/f9530e3b1ab807e0e13e5f6f26c06e48b4f46f61/spec/ddtrace/contrib/rails/disable_env_spec.rb#L21-L23

Reply:
makes sense, updated